flo_ai/flo_ai/tools/transformer_lattice_tool.py

```
@@ -0,0 +1,106 @@
+    """
+
+    # Locate / build shared library
+    current_dir = Path(__file__).resolve().parent
+    lib_path = current_dir / "liblattice.so"
+    source_path = Path(os.environ.get("LATTICE_SRC", "./transformer_lattice.c")).resolve()
+
+    if not lib_path.exists():
+        if not source_path.exists():
+            raise FileNotFoundError(
+                f"Neither shared library {lib_path} nor source file {source_path} could be found."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)